module/VuFind/src/VuFind/ILS/Driver/Alma.php

```diff
@@ -233,114 +263,121 @@ protected function getAvailabilityFromItem($item)
      *
      * @param string $id     The record id to retrieve the holdings for
      * @param array  $patron Patron data
+     * @param int    $page   The selected page number of the item paginator (if
```

Comment:
I don't have a very strong opinion on this one way or the other, but I wonder if we should consider exposing offset and limit directly here instead of abstracting things into pages for the parameter list. This might allow for richer interactions (though it might also be a problem if other ILS drivers don't support that degree of flexibility).

Reply:
Just to be sure: Do you mean calculating offset and limit values in Holds and pass them from there?

Reply:
I was actually thinking of perhaps calculating those values even higher and then passing them through Holds into the driver. I also wonder if perhaps it would be better to just pass an $options array instead of individual parameters. This would give us more control over user-specified page sizes and other future features like sort preferences (more on that later). My goal is to avoid having to duplicate a lot of math and processing logic across ILS drivers if we can do it at a higher shared level... though of course without having more examples yet, it's hard to know whether we'll end up converting offets to page numbers for some APIs. I clearly haven't thought it through very carefully, but my goal is for the driver to expose what is allowed through its getConfig call, and then for the higher level code to display appropriate options based on that.

Reply:
I now put the calculations in "RecordTab\HoldingsILS" in a method called `getPaginatorOptions()`. I think this is an appropriate place for the calculations because they are concerning the paginator that resides in the holdings ILS tab in the end. I added all possible values so that other ILS drivers could profit from them, no matter what they need. The `getPaginatorOptions()` method is called in "RecordTab\holdingsils.phtml". The resulting values are passed through as a newly introduced `$options` array until it finally arrives the `getHolding()` method in the ILS driver. I think adding other params to the array should be no problem, e. g. a possible sorting param that you mentioned.